crates/miden-tx/src/executor/exec_host.rs

```diff
@@ -120,7 +126,12 @@ where
 
     /// Returns a reference to the `tx_progress` field of this transaction host.
     pub fn tx_progress(&self) -> &TransactionProgress {
-        self.base_host.tx_progress()
+        &self.tx_progress
+    }
+
+    /// Returns a mutable reference to the `tx_progress` field of this transaction host.
+    pub(super) fn tx_progress_mut(&mut self) -> &mut TransactionProgress {
+        &mut self.tx_progress
```

Comment:
Does this need to be `pub(super)`? From what I can tell, this is used only internally in this module. In fact, we can probably get rid of this method altogether.